App/In-App Actions/URLMenuPresenter.swift

```diff
@@ -441,14 +442,75 @@ final class URLMenuPresenter: NSObject {
+                  
+                    PHPhotoLibrary.requestAuthorization({ _ in })
+
+                    let alertError: (ErrorWorkaround) -> Void = {error in
+                        let title = "Error"
+                        let message = "Failed to save"
+                        DispatchQueue.main.async {
+                            let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
+                            alert.addAction(UIAlertAction(title: "OK", style: UIAlertAction.Style.cancel, handler: nil))
+                            presentingViewController.present(alert, animated: true)
+                        }
+                    }
+                    
+                    URLSession.shared.dataTask(with: resolved) { (data: Data?, response: URLResponse?, error: Error?) in
+                        do {
+                            try data!.write(to: randomURL)
+                                if UIVideoAtPathIsCompatibleWithSavedPhotosAlbum(randomURL.relativePath) {
+                                    UISaveVideoAtPathToSavedPhotosAlbum(randomURL.relativePath, nil, nil, nil)
+                                    DispatchQueue.main.async {
+                                        let alert = UIAlertController(title: "Saved!", message: "", preferredStyle: .alert)
+                                        alert.addAction(UIAlertAction(title: "OK", style: UIAlertAction.Style.cancel, handler: nil))
+                                        presentingViewController.present(alert, animated: true)
+                                    }
+                                }
+                        } catch {
+                            return alertError(.error(error))
+                        }
+                        
+                        if let response = response as? HTTPURLResponse {
+                            if !(200...299).contains(response.statusCode) {
+                                let error = NSError(
+                                    domain: URLError.errorDomain,
+                                    code: NSURLErrorBadServerResponse,
+                                    userInfo: [
+                                        NSLocalizedDescriptionKey: "Request failed (\(response.statusCode))",
+                                        NSURLErrorFailingURLErrorKey: resolved,
+                                    ]
+                                )
+                                return alertError(.error(error))
+                            }
+                        }
+                        
+                    }.resume()
+                }))
+                }
+                actionSheet.addAction(.init(title: LocalizedString("cancel"), style: .cancel))
+            
+                presentingViewController.present(actionSheet, animated: true)
```

Comment:
haha you were spot on. It did crash on iPad.
This spun me out for a good while. I kept trying to do the popover thing for the Success/Error alert messages, rather than the actionsheet as a whole. You did say actionsheet, was my error :)